### PR TITLE
tests: system disable ssh for config defaults in gadget

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Test that config defaults for ssh service specified in the gadget are picked up for first boot snaps

### CODE REVIEW

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml [hunk 1]
```diff
@@ -116,6 +120,12 @@ execute: |
...
 
     echo "The configuration for core is applied"
-    snap get core service.rsyslog.disable|MATCH true
-    echo "And the service is masked"
-    systemctl status rsyslog.service|MATCH masked
+    snap get core "service.$SERVICE.disable" | MATCH true
+
+    if [ "$SERVICE" = ssh ]; then
+        echo "And the ssh service is disabled"
+        MATCH "SSH has been disabled by snapd system configuration" < /etc/ssh/sshd_not_to_be_run
```
Comment (mvo5) [inline tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: Will we need to undo this? I mean in restore: remove the file `/etc/ssh/sshd_not_to_be_run` ? Without the next intentional reboot will not have ssh anymore, no? Also we should probably check if ssh is really stopped, I get the suspicion that we don't actively stop it (and this test can now easily confirm that).

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml [hunk 2]
```diff
@@ -55,10 +59,10 @@ restore: |
         echo "This test needs test keys to be trusted"
         exit
     fi
-    echo "Undo the rsyslog disable"
-    systemctl unmask rsyslog.service || true
-    systemctl enable rsyslog.service || true
-    systemctl start rsyslog.service || true
+    echo "Undo the service disable"
```
Comment (mvo5) [inline tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: Would we have a "if [ "$SERVICE" = ssh ]; then"  just like we have below? One unmasks, the other removes the confi file?

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml [hunk 3]
```diff
@@ -32,7 +36,7 @@ prepare: |
          b: B
        system:
          service:
-           rsyslog:
+           $SERVICE:
```
Comment (pedronis) [inline tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: we want to test also:

${SERVICE}.disable: true

We probably want to have a 2nd test about only ssh, and then use  variants for the two layouts for the default settings there.

Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: sure, I'll make that change. Thanks for reviewing

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: thanks!! restore added

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: yes, thanks for the review

File: tests/main/ubuntu-core-gadget-config-defaults/gadget-ssh-oneline.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+defaults:
+   TEST_SNAP_ID:
+     a: A
+     b: B
+   system:
+     service:
```
Comment (pedronis) [inline tests/main/ubuntu-core-gadget-config-defaults/gadget-ssh-oneline.yaml]: as we discussed  this doesn't work but

     service.ssh.disable: true

should and we need to keep it working.

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml (hunk 1 above)
Comment (pedronis) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: > Also we should probably check if ssh is really stopped, I get the suspicion that we don't actively stop it (and this test can now easily confirm that).

this wasn't added though

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml [hunk 5]
```diff
@@ -55,10 +59,14 @@ restore: |
         echo "This test needs test keys to be trusted"
         exit
     fi
-    echo "Undo the rsyslog disable"
-    systemctl unmask rsyslog.service || true
-    systemctl enable rsyslog.service || true
-    systemctl start rsyslog.service || true
+    echo "Undo the service disable"
+    if [ "$SERVICE" = ssh ]; then
+        rm -f /etc/ssh/sshd_not_to_be_run
```
Comment (pedronis) [inline tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: should we have a "start" here too for symmetry

Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: done

File: tests/main/ubuntu-core-gadget-config-defaults/gadget-ssh-oneline.yaml (hunk 4 above)
Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/gadget-ssh-oneline.yaml]: it is includede in the last commit

File: tests/main/ubuntu-core-gadget-config-defaults/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/ubuntu-core-gadget-config-defaults/task.yaml]: Already added

**Pull request merged**